compiler/rustc_hir/src/hir_id.rs

```diff
@@ -60,14 +67,22 @@ impl<CTX: HashStableContext> ToStableHashKey<CTX> for OwnerId {
 /// the `local_id` part of the `HirId` changing, which is a very useful property in
 /// incremental compilation where we have to persist things through changes to
 /// the code base.
-#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
+#[derive(Copy, Clone, PartialEq, Eq, Hash)]
 #[derive(Encodable, Decodable, HashStable_Generic)]
 #[rustc_pass_by_value]
 pub struct HirId {
     pub owner: OwnerId,
     pub local_id: ItemLocalId,
 }
 
+impl Debug for HirId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // Example: HirId(10, OwnerId(DefId(0:1 ~ aa[7697]::{use#0})))
+        // Don't use debug_tuple to always keep this on one line.
+        write!(f, "HirId({:?}, {:?})", self.local_id, self.owner)
```

Comment:
Why in this order? I'd have put the container (owner) before the containee (local_id).
Maybe an even shorter form:
```suggestion
        write!(f, "{:?}.{:?}", self.owner, self.local_id)
```

Reply:
Doing it like this does make more sense semantically, but it puts the less interesting information first. I think it's fine to print the `OwnerId` with it even if it doesn't care much useful information since it's just trailing around and putting the `ItemLocalId` first ensures that the more relevant information is at the beginning.

But if you think that it's not worth it to show the `OwnerId` for the `HirId` at all I can also remove it.

Reply:
Showing `self.owner` is definitely useful. I adapted my other comment to make it a bit more clear.

> putting the ItemLocalId first ensures that the more relevant information is at the beginning.

This depends what you are trying to debug. In doubt, the semantic order is a good default.